framework/include/materials/MaterialPropertyInterface.h

```diff
@@ -164,6 +165,8 @@ class MaterialPropertyInterface
   bool getMaterialPropertyCalled() const { return _get_material_property_called; }
 
 protected:
+  /// Parameters of the object with this interface
+  const InputParameters & _mi_params;
```

Comment:
Maybe, but there are two reasons I'm not doing that yet. First, it combines multiple actions into one PR and second, it may just plain not work! We have unique names for these class members because several of our concrete classes inherit from multiple interfaces. If we do this in the new base class we may have ambiguous name issues. I think we may get away with it if we change all of the access levels to private. In either case I'll put in a second PR for all of that.